pkg/apis/certmanager/v1alpha1/types_issuer.go

```diff
@@ -274,6 +275,19 @@ type ACMEIssuerDNS01ProviderRFC2136 struct {
 	TSIGAlgorithm string `json:"tsigAlgorithm"`
 }
 
+type ACMEIssuerDNS01ProviderWebhook struct {
+	URL string `json:"url"`
+
+	// + optional
+	Metadata map[string]string `json:"metadata"`
```

Comment:
Should have `omitempty` else there could be `null` values output here (all maps are pointers in go)